fake-lmp-device-register

```diff
@@ -50,19 +56,35 @@ extendedKeyUsage=critical, clientAuth
 """
         )
         cnf.flush()
-
-        r = subprocess.run(
-            ["openssl", "req", "-new", "-config", cnf.name, "-key", "-"],
-            input=key,
-            stdout=subprocess.PIPE,
-        )
-        r.check_returncode()
+        cnf.close()
+
+        r = subprocess.run(["openssl", "req", "-new", "-config", cnf.name, "-key", keyfile_path], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if r.returncode != 0:
+            log.error("Error in running openssl command:")
+            log.error(r.stderr.decode())
+            os.unlink(cnf.name)
+            os.unlink(keyfile_path)
+            raise subprocess.CalledProcessError(r.returncode, r.args)
+        os.unlink(cnf.name)
+        os.unlink(keyfile_path)
```

Comment:
looks like you are making two changes in one commit and I'm not sure I understand what keeping these temp files around is actually achieving?

Reply:
@doanac you beat me to it :-)

@camilamacedo86 could you either split this PR into several commits, or only leave what is relevant for the fix itself.
It is hard to grasp what was the problem and what is the fix.

Reply:
If we break each one of them will not work alone 
But I will try to do so

Reply:
its not so much that - it's why did you have to change this? I think this part of the code worked and was simpler to follow.

and then there's logging changes and a change to the default location it writes to.  So what is broke and what needs to change.

Reply:
This part did not work. The error was faced when we were trying to read the key from the input.
I tried other ways, but the way that I could make it work was ensuring a temp file to be used.

Error we fixed (Example):
    
    $ python3 fake-lmp-device-register --registration-url "http://localhost:80/sign" --factory fioed-camila-sample private key from -: No such file or directoryTraceback (most recent call last):  File "/Users/camilamacedo/go/src/github.com/foundriesio/factory-registration-ref/fake-lmp-device-register", line 146, in <module>    main(options)  File "/Users/camilamacedo/go/src/github.com/foundriesio/factory-registration-ref/fake-lmp-device-register", line 65, in main    pkey, csr = create_key(opts.uuid, opts.factory, opts.production)                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "/Users/camilamacedo/go/src/github.com/foundriesio/factory-registration-ref/fake-lmp-device-register", line 59, in create_key    r.check_returncode()  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 502, in check_returncode    raise CalledProcessError(self.returncode, self.args, self.stdout,subprocess.CalledProcessError: Command '['openssl', 'req', '-new', '-config', '/var/folders/zr/51746skj0gz6qnt27hmvvbnc0000gn/T/tmp5ph4z49y', '-key', '-']' returned non-zero exit status 1.

Reply:
But I will add a try catch to ensure that we delete always the temp files. 
hold on I am pushing a PR with the commits and some nit improvements